### PR TITLE
Fix Azure storage, store only information provided

### PR DESCRIPTION
**Pull request by Arkatufus**
Store host name or address as null instead of trying to infer their value, because trying to infer this value can cause a lot of problem in certain systems (ie. systems that does not have DNS resolvers)

### CODE REVIEW

File: src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs [hunk 1]
```diff
@@ -136,25 +134,17 @@ private bool Initializing(object message)
...
 
-                            _host = Dns.GetHostName();
-                        }
-                        else
-                        {
-                            _host = _settings.HostName;
-                            var addresses = Dns.GetHostAddresses(_host);
-                            _address = addresses.First(i => i.AddressFamily == AddressFamily.InterNetwork && !Equals(i, IPAddress.Any));
-                        }
+                        _host = null;
```
Comment (Arkatufus) [inline src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs:142]: Just store null if host information was not given

File: src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs [hunk 2]
```diff
@@ -136,25 +134,17 @@ private bool Initializing(object message)
...
                     }
-                    catch (Exception ex)
+                    else
                     {
-                        throw new InvalidOperationException($"Failed to invoke Dns.GetHostEntry() for host [{_host}]", ex);
+                        _host = _settings.HostName;
+                        _address = null;
```
Comment (Arkatufus) [inline src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs:147]: Just store null if IP address was not given

**Comment by Aaronontheweb** [comment]
@Arkatufus what motivated this PR? Did you run into this issue somewhere in the wild?

**Comment by Arkatufus** [comment]
Yes, in an environment that doesn't support DNS lookup, trying to infer host name or trying to retrieve IP addresses of a host using DNS will fail

**Comment by Arkatufus** [comment]
So it is a lot safer to just not do it, we don't need to have both IPAddress and host name to be populated for discovery.

File: src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs:142]: LGTM

File: src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs (hunk 2 above)
Comment (Aaronontheweb) [reply src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs:147]: lgtm

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**